翻译glTF模型和图像位置解析算法 (#1790)
cocos-creator/creator-docs: en/asset/model/image-location-resolution.md

# Cocos Creator Image Location Resolution Algorithm

The parameters given in the Cocos Creator image location resolution algorithm and their descriptions are as follows:

| Parameter | Description |
| :---- | :------ |
| `url` | The desired URL |
| `startDir` | The starting search directory |
| `DEPTH` | Search depth, fixed to **2** |
| `SEARCH_DIR_NAMES` | Array of texture folder names, default is: `textures`, `materials` |
| `SEARCH_EXT_NAMES` | Array of extensions to search for, fixed to `.jpg`, `.jpeg`, `.png`, `.tga`, `.webp` |

The Cocos Creator image location resolution algorithm is given by the following procedure:

- If the file corresponding to `url` exists, return `url`.

- Let `expectedExtName` be the extension of `url`.

> - let `expectedBaseName` be the de-extended filename of `url`.

Reviewer: de-extended -> extensionless

> - let `expectedBaseName` be the de-extended filename of `url`.

Author: ```suggestion
- let `expectedBaseName` be the extensionless filename of `url`.

```

# Cocos Creator Image Location Resolution Algorithm

> The parameters given in the Cocos Creator image location resolution algorithm and their descriptions are as follows:

Reviewer 2: 这里的 parameter 应该是 property，麻烦帮忙把中文的 “参数” 也改成 “属性”

Reviewer 2: 抱歉标记错了，是 glTF 贴图参数，已经重新标记了

> The parameters given in the Cocos Creator image location resolution algorithm and their descriptions are as follows:

Author: ```suggestion
The properties given in the Cocos Creator image location resolution algorithm and their descriptions are as follows:

```

> | Parameter | Description |

Reviewer 2: 这里的 parameter 应该是 property，麻烦帮忙把中文的 “参数” 也改成 “属性”

> | Parameter | Description |

Author: ```suggestion
| Property | Description |

```

| :---- | :------ |
| `url` | The desired URL |
| `startDir` | The starting search directory |
| `DEPTH` | Search depth, fixed to **2** |
| `SEARCH_DIR_NAMES` | Array of texture folder names, default is: `textures`, `materials` |
| `SEARCH_EXT_NAMES` | Array of extensions to search for, fixed to `.jpg`, `.jpeg`, `.png`, `.tga`, `.webp` |

The Cocos Creator image location resolution algorithm is given by the following procedure:

- If the file corresponding to `url` exists, return `url`.

- Let `expectedExtName` be the extension of `url`.

- let `expectedBaseName` be the extensionless filename of `url`.

> - Let `searchExtNames` be `[expectedExtName, ...SEARCH_EXT_NAMES]` array after de-duplication.

Reviewer 2: 这个应该是 “Let `searchExtNames` be the array after the `[expectedExtName, ...SEARCH_EXT_NAMES]` de-duplication” 比较准确些，不过语法不确定对不对

> - Let `searchExtNames` be `[expectedExtName, ...SEARCH_EXT_NAMES]` array after de-duplication.

Author: ```suggestion
- Let `searchExtNames` be the array of `[expectedExtName, ...SEARCH_EXT_NAMES]` after de-duplication.

```

- Let `currentDir` be `startDir` and loop for `DEPTH` times:

>     - If `currentDir` is outside the project `assets` directory, then exit the loop.

Reviewer 2: ```suggestion
    - If `currentDir` is outside the project's `assets` directory, then exit the loop.

```
